src/Dialogs/AvatarDialog.vala

```diff
@@ -41,18 +41,11 @@ public class SwitchboardPlugUserAccounts.Dialogs.AvatarDialog : Granite.MessageD
         response.connect (on_response);
 
         try {
-            cropview = new Widgets.CropView.from_pixbuf_with_size (new Gdk.Pixbuf.from_file (pixbuf_path), 400, 300);
+            Gdk.Pixbuf pixbuf = new Gdk.Pixbuf.from_file (pixbuf_path).apply_embedded_orientation ();
+            cropview = new Widgets.CropView.from_pixbuf_with_size (pixbuf, 400, 400);
             cropview.quadratic_selection = true;
             cropview.handles_visible = false;
-
-            var frame = new Gtk.Grid ();
-            frame.add (cropview);
-
-            var frame_context = frame.get_style_context ();
-            frame_context.add_class (Granite.STYLE_CLASS_CARD);
-            frame_context.add_class (Granite.STYLE_CLASS_CHECKERBOARD);
-
-            custom_bin.add (frame);
```

Comment:
This is actually important for both applying shadows and also for making sure this area has a background for smaller or semi-transparent images

Reply:
Ah, I missed the shadows! And I hadn't considered semi-transparent images. The reason I removed this part was that with the image in the upright orientation, a small sliver of checkerboard would be visible, which I found less nice:

![cropview-with-checkerboard](https://user-images.githubusercontent.com/12174852/227859121-273b6d0c-5bc9-4d28-9bf0-e98ad665e2ed.png)

But I'll restore the old behavior in a follow up commit; thanks.

Reply:
Ah yeah I think we might have fixed that since we're making sure to use the preview size for the image size. At least in my testing for larger images it seems to go to the edge